evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -170,14 +144,22 @@ def fit(self, X, y=None):
         Raises:
             ValueError: If X was passed to `fit` but not passed in `predict`.
         """
+        X, y = self._manage_woodwork(X, y)
         if y is None:
             raise ValueError("ARIMA Regressor requires y as input.")
+        if X is not None:
+            target_leakage = TargetLeakageDataCheck().validate(X, y)["actions"]
+            leaked_columns = (
+                target_leakage[0]["metadata"]["columns"] if target_leakage else []
+            )
+            self.cols_to_keep = list(set(X.columns) - set(leaked_columns))
+            X = X[self.cols_to_keep]
```

Comment:
So the previous implementation dropped `datetime` columns because ARIMA doesn't accept those, so this is an extension of that. I totally understand the trepidation over this, but since this isn't a `Transformer` we luckily won't be passing the features onto anything else.

That being said, I do think that maybe a warning should be added to the logs, is that along the lines of what you were thinking?

Reply:
I would support adding something to logs! Even if it's not at the `WARNING` level, just a debug message could be better than nothing.